folium/plugins/timestamped_geo_json.py

```diff
@@ -133,6 +137,7 @@ def __init__(self, data, transition_time=200, loop=True, auto_play=True, add_las
         self.add_last_point = bool(add_last_point)
         self.period = period
         self.date_options = date_options
+        self.duration = duration
```

Comment:
Then this could be `self.duration = duration if duration is not None else 'undefined'`